scrapy_zyte_api/providers.py

```diff
@@ -112,10 +125,35 @@ async def __call__(  # noqa: C901
                     options["extractFrom"] = extract_from.value
                     break
 
+        http_response_needed = (
+            AnyResponse in to_provide
+            and BrowserResponse not in to_provide
+            and BrowserHtml not in to_provide
+            and not http_response
+        )
+
+        extract_from = None  # type: ignore[assignment]
         for item_type, kw in item_keywords.items():
             options_name = f"{kw}Options"
             if item_type not in to_provide_stripped and options_name in zyte_api_meta:
                 del zyte_api_meta[options_name]
+            elif options_name in zyte_api_meta:
+                extract_from = zyte_api_meta[options_name]["extractFrom"]
+            elif item_type in to_provide_stripped and http_response_needed:
+                zyte_api_meta[options_name] = {"extractFrom": "httpResponseBody"}
+
+        if AnyResponse in to_provide:
+            if extract_from == "browserHtml":
+                html_requested = True
+            elif extract_from == "httpResponseBody" or http_response_needed:
+                param_parser = _ParamParser(crawler)
+                param_parser._transparent_mode = True
+                http_request_params = param_parser.parse(request)
+                del http_request_params["url"]
+                zyte_api_meta.update(http_request_params)
```

Comment:
I think we may need to use this above for browser as well, if the goal is to map things like `requestHeaders`.

I also wonder if it might be safer to run `param_parser._automap_params = zyte_api_meta` than to merge the dicts later, but I think at the moment they might be functionally equivalent

Reply:
I've added a TODO comment about mapping `requestHeaders` for `browserHtml` for the future. I'm worried that it might cause some regressions if introduced in this PR.

Thanks for the `param_parser._automap_params = zyte_api_meta` tip. I tried this but there's a few tests that fails if I use it. In particular, some zyte parameters will be missing.